File: tests/Web.AcceptanceTests/Pages/CalculatorsPage.cs
namespace LiftAndShift.Web.AcceptanceTests.Pages;

public class CalculatorsPage(IPage page) : BasePage(page)
{
    public override string PagePath => $"{BaseUrl}/calculators";

    public Task AssertHeading() => Assertions.Expect(Page.Locator("h1")).ToHaveTextAsync("Calculators");

    public Task ClickWarmupCalculatorTab() => Page.Locator("button:has-text('Warmup Calculator')").ClickAsync();

    public Task SetPlateTargetWeight(string value) => Page.Locator("section form input[type='number']").First.FillAsync(value);

    public Task SetWarmupWorkingWeight(string value) => Page.Locator("section form input[type='number']").First.FillAsync(value);

    public Task ClickCalculate() => Page.Locator("button[type='submit']:has-text('Calculate')").ClickAsync();

    public Task AssertPlateResultVisible() => Assertions.Expect(Page.Locator("text=Actual weight:")).ToBeVisibleAsync();

    public Task AssertWarmupTableVisible() => Assertions.Expect(Page.Locator("table")).ToBeVisibleAsync();
}

File: src/Web/ClientApp/e2e/pages/calculators.page.ts
import { expect } from '@playwright/test';
import { BasePage, baseUrl } from './base-page';

export class CalculatorsPage extends BasePage {
  get pagePath(): string {
    return `${baseUrl()}/calculators`;
  }

  assertHeading() {
    return expect(this.page.locator('h1')).toHaveText('Calculators');
  }

  clickWarmupCalculatorTab() {
    return this.page.locator("button:has-text('Warmup Calculator')").click();
  }

  setPlateTargetWeight(value: string) {
    return this.page.locator("section form input[type='number']").first().fill(value);
  }

  setWarmupWorkingWeight(value: string) {
    return this.page.locator("section form input[type='number']").first().fill(value);
  }

  clickCalculate() {
    return this.page.locator("button[type='submit']:has-text('Calculate')").click();
  }

  assertPlateResultVisible() {
    return expect(this.page.locator('text=Actual weight:')).toBeVisible();
  }

  assertWarmupTableVisible() {
    return expect(this.page.locator('table')).toBeVisible();
  }
}
